File: U1/PROYECTO/Comercializadora_Restful_Java_GR01/src/main/java/ec/edu/monster/dto/ProductoDTO.java
package ec.edu.monster.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * DTO para transferencia de datos de Producto
 */
public class ProductoDTO {
    
    private Integer productoId;
    private String codigo;
    private String nombre;
    private String descripcion;
    private BigDecimal precio;
    private Integer stock;
    private String categoria;
    private String imagenUrl;
    private LocalDateTime fechaRegistro;
    private String estado;
    
    public ProductoDTO() {
    }
    
    public ProductoDTO(Integer productoId, String codigo, String nombre, String descripcion, 
                        BigDecimal precio, Integer stock, String categoria, String imagenUrl, 
                        LocalDateTime fechaRegistro, String estado) {
        this.productoId = productoId;
        this.codigo = codigo;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.stock = stock;
        this.categoria = categoria;
        this.imagenUrl = imagenUrl;
        this.fechaRegistro = fechaRegistro;
        this.estado = estado;
    }
    
    public Integer getProductoId() {
        return productoId;
    }
    
    public void setProductoId(Integer productoId) {
        this.productoId = productoId;
    }
    
    public String getCodigo() {
        return codigo;
    }
    
    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }
    
    public String getNombre() {
        return nombre;
    }
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public String getDescripcion() {
        return descripcion;
    }
    
    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
    
    public BigDecimal getPrecio() {
        return precio;
    }
    
    public void setPrecio(BigDecimal precio) {
        this.precio = precio;
    }
    
    public Integer getStock() {
        return stock;
    }
    
    public void setStock(Integer stock) {
        this.stock = stock;
    }
    
    public String getCategoria() {
        return categoria;
    }
    
    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }
    
    public String getImagenUrl() {
        return imagenUrl;
    }
    
    public void setImagenUrl(String imagenUrl) {
        this.imagenUrl = imagenUrl;
    }
    
    public LocalDateTime getFechaRegistro() {
        return fechaRegistro;
    }
    
    public void setFechaRegistro(LocalDateTime fechaRegistro) {
        this.fechaRegistro = fechaRegistro;
    }
    
    public String getEstado() {
        return estado;
    }
    
    public void setEstado(String estado) {
        this.estado = estado;
    }
}

File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/Comercializadora_Soap_DotNet_GR01/Comercializadora_Soap_DotNet_GR01/DTOs/ProductoDTO.cs
using System;
using System.Runtime.Serialization;

namespace Comercializadora_Soap_DotNet_GR01.DTOs
{
    [DataContract]
    public class ProductoDTO
    {
        [DataMember(Order = 1)]
        public int ProductoId { get; set; }

        [DataMember(Order = 2)]
        public string Codigo { get; set; }

        [DataMember(Order = 3)]
        public string Nombre { get; set; }

        [DataMember(Order = 4)]
        public string Descripcion { get; set; }

        [DataMember(Order = 5)]
        public decimal Precio { get; set; }

        [DataMember(Order = 6)]
        public int Stock { get; set; }

        [DataMember(Order = 7)]
        public string Categoria { get; set; }

        [DataMember(Order = 8)]
        public string ImagenUrl { get; set; }

        [DataMember(Order = 9)]
        public DateTime FechaRegistro { get; set; }

        [DataMember(Order = 10)]
        public string Estado { get; set; }
    }
}